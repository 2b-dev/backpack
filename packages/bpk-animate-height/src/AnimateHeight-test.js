import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import AnimateHeight from './AnimateHeight';

describe('AnimateHeight', () => {
  it('should render correctly with "height" attribute equal to "auto"', () => {
    const tree = renderer.create(
      <AnimateHeight
        duration={200}
        height="auto"
      >
        Content.
      </AnimateHeight>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with "height" attribute equal to "200"', () => {
    const tree = renderer.create(
      <AnimateHeight
        duration={200}
        height={200}
      >
        Content.
      </AnimateHeight>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should set "display: none;" on contentElement if height is 0', () => {
    const tree = mount(
      <AnimateHeight
        duration={0}
        height={0}
      >
        Content.
      </AnimateHeight>,
      { lifecycleExperimental: true }, // See https://github.com/airbnb/enzyme/pull/318
    );

    expect(tree.instance().contentElement.style.display).toEqual('none');

    tree.setProps({ height: 100 }).update();

    expect(tree.instance().contentElement.style.display).toEqual('');
  });
});
