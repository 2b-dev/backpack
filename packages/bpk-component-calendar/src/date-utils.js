import startOfMonth from 'date-fns/start_of_month';
import endOfMonth from 'date-fns/end_of_month';
import lastDayOfMonth from 'date-fns/last_day_of_month';
import startOfWeek from 'date-fns/start_of_week';
import lastDayOfWeek from 'date-fns/last_day_of_week';
import startOfToday from 'date-fns/start_of_today';
import eachDay from 'date-fns/each_day';
import getDay from 'date-fns/get_day';

import isWithinRange from 'date-fns/is_within_range';
import isToday from 'date-fns/is_today';
import isSaturday from 'date-fns/is_saturday';
import isSunday from 'date-fns/is_sunday';
import isSameDay from 'date-fns/is_same_day';
import isSameMonth from 'date-fns/is_same_month';
import isBefore from 'date-fns/is_before';

import addDays from 'date-fns/add_days';
import addWeeks from 'date-fns/add_weeks';
import addMonths from 'date-fns/add_months';
import setMonth from 'date-fns/set_month';
import setYear from 'date-fns/set_year';

import parse from 'date-fns/parse';
import format from 'date-fns/format';

function getCalendarMonthWeeks(month, weekStartsOn) {
  const baseDate = new Date(month);
  const firstOfMonth = startOfMonth(baseDate);

  let currentDay = startOfWeek(firstOfMonth, { weekStartsOn });
  const weeksInMonth = [];

  for (let i = 0; i < 6; i += 1) {
    const currentWeek = eachDay(currentDay, lastDayOfWeek(currentDay, { weekStartsOn }));
    weeksInMonth.push(currentWeek);
    currentDay = addWeeks(currentDay, 1);
  }

  return weeksInMonth;
}

/* Takes arbitrary dates and returns the beginning of the first and enf of the last month containing these dates  */
function getMonthRange(from, to) {
  return {
    min: startOfMonth(from),
    max: endOfMonth(to),
  };
}

function getMonthsInRange(from, to) {
  const { min, max } = getMonthRange(from, to);
  let currentMonth = startOfMonth(from);
  const monthsInRange = [];

  while (isWithinRange(currentMonth, min, max)) {
    monthsInRange.push(currentMonth);
    currentMonth = addMonths(currentMonth, 1);
  }

  return monthsInRange;
}

const parseIsoDate = parse;
const formatIsoDate = date => format(date, 'YYYY-MM-DD');
const formatIsoMonth = date => format(date, 'YYYY-MM');

export {
  getCalendarMonthWeeks,
  getMonthsInRange,
  getMonthRange,
  getDay,
  isWithinRange,
  isSaturday,
  isSunday,
  isToday,
  isSameDay,
  isSameMonth,
  isBefore,
  addMonths,
  addDays,
  setMonth,
  setYear,
  startOfMonth,
  lastDayOfMonth,
  startOfToday,
  format,
  formatIsoDate,
  formatIsoMonth,
  parseIsoDate,
};
