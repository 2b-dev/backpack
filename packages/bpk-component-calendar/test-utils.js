import format from 'date-fns/format';

export const formatDateFull = date => format(date, 'dddd, Do MMMM YYYY');
export const formatMonth = date => format(date, 'MMMM YYYY');
export const formatMonthGerman = (date) => {
  const months = [
    'Januar',
    'Februar',
    'MÃ¤rz',
    'April',
    'Mai',
    'Juni',
    'Juli',
    'August',
    'September',
    'Oktober',
    'November',
    'Dezember',
  ];
  return `${months[date.getMonth()]} ${date.getUTCFullYear()}`;
};

export const weekDays = [
  {
    name: 'Sunday',
    nameAbbr: 'Sun',
    index: 0,
    isWeekend: true,
  },
  {
    name: 'Monday',
    nameAbbr: 'Mon',
    index: 1,
    isWeekend: false,
  },
  {
    name: 'Tuesday',
    nameAbbr: 'Tue',
    index: 2,
    isWeekend: false,
  },
  {
    name: 'Wednesday',
    nameAbbr: 'Wed',
    index: 3,
    isWeekend: false,
  },
  {
    name: 'Thursday',
    nameAbbr: 'Thu',
    index: 4,
    isWeekend: false,
  },
  {
    name: 'Friday',
    nameAbbr: 'Fri',
    index: 5,
    isWeekend: false,
  },
  {
    name: 'Saturday',
    nameAbbr: 'Sat',
    index: 6,
    isWeekend: true,
  },
];

export const weekDaysMoreWeekend = [
  {
    name: 'Sunday',
    nameAbbr: 'Sun',
    index: 0,
    isWeekend: true,
  },
  {
    name: 'Monday',
    nameAbbr: 'Mon',
    index: 1,
    isWeekend: false,
  },
  {
    name: 'Tuesday',
    nameAbbr: 'Tue',
    index: 2,
    isWeekend: false,
  },
  {
    name: 'Wednesday',
    nameAbbr: 'Wed',
    index: 3,
    isWeekend: false,
  },
  {
    name: 'Thursday',
    nameAbbr: 'Thu',
    index: 4,
    isWeekend: false,
  },
  {
    name: 'Friday',
    nameAbbr: 'Fri',
    index: 5,
    isWeekend: true,
  },
  {
    name: 'Saturday',
    nameAbbr: 'Sat',
    index: 6,
    isWeekend: true,
  },
];

export const weekDaysGerman = [
  {
    name: 'Sonntag',
    nameAbbr: 'So.',
    index: 0,
    isWeekend: true,
  },
  {
    name: 'Montag',
    nameAbbr: 'Mo.',
    index: 1,
    isWeekend: false,
  },
  {
    name: 'Dienstag',
    nameAbbr: 'Di.',
    index: 2,
    isWeekend: false,
  },
  {
    name: 'Mittwoch',
    nameAbbr: 'Mi.',
    index: 3,
    isWeekend: false,
  },
  {
    name: 'Donnerstag',
    nameAbbr: 'Do.',
    index: 4,
    isWeekend: false,
  },
  {
    name: 'Freitag',
    nameAbbr: 'Fr.',
    index: 5,
    isWeekend: false,
  },
  {
    name: 'Samstag',
    nameAbbr: 'Sa.',
    index: 6,
    isWeekend: true,
  },
];
