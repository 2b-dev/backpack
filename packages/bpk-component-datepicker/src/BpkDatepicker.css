@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-datepicker__input{text-align:inherit}.bpk-datepicker__input--placeholder{color:#68697f;font-style:italic}.bpk-datepicker__calendar{margin:0 auto}
