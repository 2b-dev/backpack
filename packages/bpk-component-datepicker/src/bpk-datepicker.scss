@import "~bpk-mixins";

.bpk-datepicker {
  &__input {
    cursor: pointer;

    &-description {
      @include bpk-visually-hidden();
    }
  }
}
