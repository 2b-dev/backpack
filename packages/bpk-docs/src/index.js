import React from 'react';
import ReactDOM from 'react-dom';
import Helmet from 'react-helmet';
import ReactDOMServer from 'react-dom/server';
import { Router, RouterContext, match, browserHistory, createMemoryHistory } from 'react-router';

import routes from './routes';
import template from './template';
import { extractAssets } from './webpackStats';

if (typeof document !== 'undefined') {
  const root = document.getElementById('react-mount');

  ReactDOM.render(React.createElement(Router, { history: browserHistory, routes }), root);
}

export default ((locals, callback) => {
  const history = createMemoryHistory();
  const location = history.createLocation(locals.path);
  const assets = extractAssets(locals.webpackStats);

  match({ routes, location, history }, (error, redirectLocation, props) => {
    if (error) {
      return callback(error);
    }

    if (redirectLocation) {
      return callback(error, `<script>window.location = '${redirectLocation.pathname}';</script>`);
    }

    const head = Helmet.rewind();
    const html = ReactDOMServer.renderToStaticMarkup(React.createElement(RouterContext, props));

    return callback(error, template({ head, html, assets }));
  });
});
