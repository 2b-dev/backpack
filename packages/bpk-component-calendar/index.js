import withCalendarState from './src/BpkCalendarContainer';
import BpkCalendarGrid from './src/BpkCalendarGrid';
import BpkCalendarGridHeader from './src/BpkCalendarGridHeader';
import BpkCalendarNav from './src/BpkCalendarNav';
import BpkCalendarDate from './src/BpkCalendarDate';
import BpkCalendar, { composeCalendar } from './src/BpkCalendar';
import CustomPropTypes from './src/custom-proptypes';
import * as DateUtils from './src/date-utils';

export default withCalendarState(BpkCalendar);

export {
  BpkCalendarGrid,
  BpkCalendarGridHeader,
  BpkCalendarNav,
  BpkCalendarDate,
  CustomPropTypes,
  DateUtils,
  withCalendarState,
  composeCalendar,
};
