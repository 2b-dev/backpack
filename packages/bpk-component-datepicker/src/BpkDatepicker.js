import BpkInput from 'bpk-component-input';
import BpkModal from 'bpk-component-modal';
import BpkPopover from 'bpk-component-popover';
import React, { PropTypes, Component } from 'react';
import BpkBreakpoint, { BREAKPOINTS } from 'bpk-component-breakpoint';
import BpkCalendar, { CustomPropTypes } from 'bpk-component-calendar';

import './bpk-datepicker.scss';

const KEYCODES = {
  ENTER: 13,
  SPACEBAR: 32,
};

const handleKeyEvent = (keyCode, callback) => (e) => {
  if (e.keyCode === keyCode) {
    e.preventDefault();
    callback();
  }
};

const withEventHandler = (fn, eventHandler) => (e) => {
  fn(e);
  if (eventHandler) {
    eventHandler(e);
  }
};

class BpkDatepicker extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
    };

    this.onOpen = this.onOpen.bind(this);
    this.onClose = this.onClose.bind(this);
    this.handleDateSelect = this.handleDateSelect.bind(this);
    this.handleTouchEnd = this.handleTouchEnd.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleBlur = this.handleBlur.bind(this);

    this.focusCanOpen = true;
  }

  onOpen() {
    this.setState({
      isOpen: true,
    });
  }

  onClose() {
    this.setState({
      isOpen: false,
    });
  }

  handleDateSelect(dateObj) {
    this.setState({
      isOpen: false,
    });
    if (this.props.onDateSelect) {
      this.props.onDateSelect(dateObj);
    }
  }

  handleTouchEnd(event) {
    // preventDefault fixes an issue on Android and iOS in which the popover closes immediately
    // because a touch event is registered on one of the dates.
    // We can only run preventDefault when the input is already focused - otherwise it would never set
    // focus on it, and when closing the modal/popover focus would return to the previously focused
    // element (which is annoying if it's an autosuggest or another datepicker, for example).
    if (document && event.target === document.activeElement) {
      event.preventDefault();
      this.onOpen();
    }
  }

  handleFocus() {
    if (this.focusCanOpen) {
      this.onOpen();
    }
  }

  handleBlur() {
    // If the input loses focus when the popover/modal is open, it should not open on a subsequent focus.
    // Fixes an issue with IE9.
    if (this.state.isOpen) {
      this.focusCanOpen = false;
    } else {
      this.focusCanOpen = true;
    }
  }

  render() {
    const {
      changeMonthLabel,
      closeButtonText,
      DateComponent,
      date,
      dateModifiers,
      daysOfWeek,
      formatDate,
      formatDateFull,
      formatMonth,
      getApplicationElement,
      id,
      inputProps,
      markOutsideDays,
      markToday,
      maxDate,
      minDate,
      onMonthChange,
      showWeekendSeparator,
      title,
      weekStartsOn,
      hasTouchSupport,
      ...rest
    } = this.props;

    const {
      onClick,
      onFocus,
      onBlur,
      onTouchEnd,
      onKeyDown,
      onKeyUp,
      ...restInputProps
    } = inputProps;

    // The following props are not used in render
    delete rest.onDateSelect;

    const eventHandlers = {
      onClick: withEventHandler(this.onOpen, onClick),
      onKeyDown: withEventHandler(handleKeyEvent(KEYCODES.ENTER, this.onOpen), onKeyDown),
      onKeyUp: withEventHandler(handleKeyEvent(KEYCODES.SPACEBAR, this.onOpen), onKeyUp),
    };

    if (hasTouchSupport) {
      // Prevents the mobile keyboard from opening (iOS / Android)
      eventHandlers.readOnly = 'readOnly';
      eventHandlers.onTouchEnd = withEventHandler(this.handleTouchEnd, onTouchEnd);
    } else {
      eventHandlers.onFocus = withEventHandler(this.handleFocus, onFocus);
      eventHandlers.onBlur = withEventHandler(this.handleBlur, onBlur);
    }

    const inputComponent = (
      <BpkInput
        id={id}
        name={`${id}_input`}
        value={date ? formatDate(date) : ''}
        className="bpk-datepicker__input"
        aria-live="assertive"
        aria-atomic="true"
        aria-label={formatDateFull(date)}
        onChange={() => null}
        {...eventHandlers}
        {...restInputProps}
      />
    );

    const calendarComponent = (
      <BpkCalendar
        className="bpk-datepicker__calendar"
        changeMonthLabel={changeMonthLabel}
        DateComponent={DateComponent}
        date={date}
        dateModifiers={dateModifiers}
        daysOfWeek={daysOfWeek}
        formatDateFull={formatDateFull}
        formatMonth={formatMonth}
        id={`${id}-calendar`}
        initialSelectedDate={this.state.date}
        markOutsideDays={markOutsideDays}
        markToday={markToday}
        maxDate={maxDate}
        minDate={minDate}
        onDateSelect={this.handleDateSelect}
        onMonthChange={onMonthChange}
        showWeekendSeparator={showWeekendSeparator}
        weekStartsOn={weekStartsOn}
      />
    );

    return (
      <BpkBreakpoint query={BREAKPOINTS.MOBILE}>
        {isMobile => (
          isMobile
            ? (
              <BpkModal
                id={`${id}-modal`}
                target={inputComponent}
                onClose={this.onClose}
                isOpen={this.state.isOpen}
                title={title}
                closeLabel={closeButtonText}
                getApplicationElement={getApplicationElement}
              >
                {calendarComponent}
              </BpkModal>
            )
            : (
              <BpkPopover
                id={`${id}-popover`}
                target={inputComponent}
                onClose={this.onClose}
                isOpen={this.state.isOpen}
                label={title}
                closeButtonText={closeButtonText}
                tabIndex={0}
                {...rest}
              >
                {calendarComponent}
              </BpkPopover>
            )
        )}
      </BpkBreakpoint>
    );
  }
}

BpkDatepicker.propTypes = {
  // Required
  changeMonthLabel: PropTypes.string.isRequired,
  closeButtonText: PropTypes.string.isRequired,
  daysOfWeek: CustomPropTypes.DaysOfWeek.isRequired,
  formatDate: PropTypes.func.isRequired,
  formatDateFull: PropTypes.func.isRequired,
  formatMonth: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  getApplicationElement: PropTypes.func.isRequired,
  // Optional
  date: PropTypes.instanceOf(Date),
  DateComponent: PropTypes.func,
  dateModifiers: CustomPropTypes.DateModifiers,
  hasTouchSupport: PropTypes.bool,
  inputProps: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  markOutsideDays: PropTypes.bool,
  markToday: PropTypes.bool,
  maxDate: PropTypes.instanceOf(Date),
  minDate: PropTypes.instanceOf(Date),
  onDateSelect: PropTypes.func,
  onMonthChange: PropTypes.func,
  showWeekendSeparator: PropTypes.bool,
  weekStartsOn: PropTypes.number,
};

BpkDatepicker.defaultProps = {
  date: null,
  DateComponent: BpkCalendar.defaultProps.DateComponent,
  dateModifiers: BpkCalendar.defaultProps.dateModifiers,
  hasTouchSupport: (typeof window !== 'undefined' && (('ontouchstart' in window) || window.DocumentTouch && document instanceof DocumentTouch)), // eslint-disable-line
  inputProps: {},
  markOutsideDays: BpkCalendar.defaultProps.markOutsideDays,
  markToday: BpkCalendar.defaultProps.markToday,
  maxDate: BpkCalendar.defaultProps.maxDate,
  minDate: BpkCalendar.defaultProps.minDate,
  onDateSelect: null,
  onMonthChange: null,
  showWeekendSeparator: BpkCalendar.defaultProps.showWeekendSeparator,
  weekStartsOn: BpkCalendar.defaultProps.weekStartsOn,
};

export default BpkDatepicker;
