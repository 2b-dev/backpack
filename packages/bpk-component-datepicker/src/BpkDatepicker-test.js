import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { weekDays, formatMonth, formatDateFull } from 'bpk-component-calendar/test-utils';
import { format } from 'bpk-component-calendar/src/date-utils';

import BpkDatepicker from './BpkDatepicker';

const formatDate = date => format(date, 'DD/MM/YYYY');

const inputProps = {
  onChange: () => null,
  placeholder: 'placeholder',
  large: true,
};

describe('BpkDatepicker', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkDatepicker
        id="myDatepicker"
        closeButtonText="Close"
        daysOfWeek={weekDays}
        changeMonthLabel="Change month"
        title="Departure date"
        getApplicationElement={() => document.createElement('div')}
        formatDate={formatDate}
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        inputProps={inputProps}
        minDate={new Date(2010, 1, 15)}
        maxDate={new Date(2010, 2, 15)}
        date={new Date(2010, 1, 15)}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should open on click', () => {
    const datepicker = mount(<BpkDatepicker
      id="myDatepicker"
      closeButtonText="Close"
      daysOfWeek={weekDays}
      changeMonthLabel="Change month"
      title="Departure date"
      getApplicationElement={() => document.createElement('div')}
      formatDate={formatDate}
      formatMonth={formatMonth}
      formatDateFull={formatDateFull}
      inputProps={inputProps}
      minDate={new Date(2010, 1, 15)}
      maxDate={new Date(2010, 2, 15)}
      date={new Date(2010, 1, 15)}
    />);

    expect(datepicker.state('isOpen')).toEqual(false);

    datepicker.find('BpkInput').simulate('click');
    expect(datepicker.state('isOpen')).toEqual(true);
  });

  it('should open on focus', () => {
    const datepicker = mount(<BpkDatepicker
      id="myDatepicker"
      closeButtonText="Close"
      daysOfWeek={weekDays}
      changeMonthLabel="Change month"
      title="Departure date"
      getApplicationElement={() => document.createElement('div')}
      formatDate={formatDate}
      formatMonth={formatMonth}
      formatDateFull={formatDateFull}
      inputProps={inputProps}
      minDate={new Date(2010, 1, 15)}
      maxDate={new Date(2010, 2, 15)}
      date={new Date(2010, 1, 15)}
    />);

    expect(datepicker.state('isOpen')).toEqual(false);

    datepicker.find('BpkInput').simulate('focus');
    expect(datepicker.state('isOpen')).toEqual(true);
  });

  it('should update state when a date is selected', () => {
    const datepicker = mount(<BpkDatepicker
      id="myDatepicker"
      closeButtonText="Close"
      daysOfWeek={weekDays}
      changeMonthLabel="Change month"
      title="Departure date"
      getApplicationElement={() => document.createElement('div')}
      formatDate={formatDate}
      formatMonth={formatMonth}
      formatDateFull={formatDateFull}
      inputProps={inputProps}
      minDate={new Date(2010, 1, 15)}
      maxDate={new Date(2010, 2, 15)}
      date={new Date(2010, 1, 15)}
    />);

    datepicker.find('BpkInput').simulate('click');
    expect(datepicker.state('isOpen')).toEqual(true);

    const date = new Date(2010, 1, 15);
    datepicker.instance().handleDateSelect(date);
    expect(datepicker.state('isOpen')).toEqual(false);
  });

  it('should close when `onClose` is called', () => {
    const datepicker = mount(<BpkDatepicker
      id="myDatepicker"
      closeButtonText="Close"
      daysOfWeek={weekDays}
      changeMonthLabel="Change month"
      title="Departure date"
      getApplicationElement={() => document.createElement('div')}
      formatDate={formatDate}
      formatMonth={formatMonth}
      formatDateFull={formatDateFull}
      inputProps={inputProps}
      minDate={new Date(2010, 1, 15)}
      maxDate={new Date(2010, 2, 15)}
      date={new Date(2010, 1, 15)}
    />);

    datepicker.find('BpkInput').simulate('click');
    expect(datepicker.state('isOpen')).toEqual(true);

    datepicker.instance().onClose();
    expect(datepicker.state('isOpen')).toEqual(false);
  });
});
