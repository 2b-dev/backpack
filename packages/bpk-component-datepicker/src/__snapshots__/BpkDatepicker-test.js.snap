exports[`BpkDatepicker should render correctly 1`] = `
<input
  aria-atomic="true"
  aria-invalid={false}
  aria-label="Monday, 15th February 2010"
  aria-live="assertive"
  className="bpk-input bpk-input--large bpk-datepicker__input"
  id="myDatepicker"
  name="myDatepicker_input"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onTouchEnd={[Function]}
  placeholder="placeholder"
  type="text"
  value="15/02/2010" />
`;
